release/models/route-programming/openconfig-route-programming-state.yang

```diff
@@ -0,0 +1,186 @@
+module openconfig-route-programming-state {
+  prefix "oc-rtstate";
+
+  namespace "http://openconfig.net/yang/route-programming";
+
+  import openconfig-extensions { prefix "oc-ext"; }
+  import openconfig-network-instance { prefix "oc-ni"; }
+  import openconfig-inet-types { prefix "oc-inet"; }
+  import openconfig-types { prefix "oc-types"; }
+
+  organization
+    "OpenConfig working group";
+
+  contact
+    "OpenConfig working group
+    www.openconfig.net";
+
+  description
+    "This module provides detailed information as to the programming
+    state of routes within a particular network instance. It can be used to
+    track where there are routing programming errors on a device.";
+
+  oc-ext:regexp-posix;
+  oc-ext:catalog-organization "openconfig";
+  oc-ext:origin "openconfig";
+
+  oc-ext:openconfig-version "0.1.0";
+
+  revision "2021-10-16" {
+    description
+      "Initial revision.";
+    reference "0.1.0";
+  }
+
+  grouping route-state-top {
+    description
+      "Top-level grouping for route state in OpenConfig.";
+
+    container route-programming {
+      config false;
+      description
+        "The programming status of routes within the network-instance. Route
+        programming reports are divided into a set of sub-categories:
+
+         - failed programming -- these routes are those that were attempted to be
+           programed to the hardware, but the programming operation failed.
+         - stale programming -- these routes are those that are programmed into
+           hardware, but some failure occurred in updating to a new state.
+         - drop programming - these routes are explicitly programmed into hardware
+           to point to a destination that discards packets.";
+
+      container failed-routes {
+        description
+          "Surrounding container for the list of routes that fail programming.";
+
+        list failed {
+          key "prefix";
+
+          description
+            "A prefix that was attempted to the programmed into hardware, but the
```

Comment:
I"m trying to understand the difference between `failed-routes` and `stale-routes`.

Is this PR trying to say the following?
* If a prefix is **not** in the FIB and being programmed into FIB but failed, categorize it into `failed-routes`. Traffic will **not** be matched to the prefix in FIB, because it's not installed yet into the FIB.
* If a prefix is already in the FIB and being updated but failed, categorize it into `stale-routes`. Traffic will still be following the *old* next-hop for forwarding. The AFT streaming will still populate the **old** next-hop for the prefix.

Reply:
Yes, your wording is perhaps more obvious